src/rain.c

```diff
@@ -0,0 +1,469 @@
+/*
+ * This software is Copyright (c) 2018 magnum
+ * and is hereby released to the general public under the following terms:
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted.
+ */
+
+/*
+ * TODO:
+ * - Profile with Callgrind
+ * - Assure hybrid support (mask or external)
+ * - Reject other hybrid modes?
+ * - Try inlining utf32_to_enc
+ * - Are we still using an unnecessary step of indexing?
+ *
+ * IDEAS:
+ * - Store charset in target encoding for quicker conversion, even for
+ *   UTF-8 (we can store it in uint32_t)! Beware of endianness.
+ *
+ * RELATED:
+ * - Unicode ranges charsets. Generator? Standalone? --subsets-file=FILE?
+ * - Add global options --skip-odd-lengths and --skip-even-lengths (also
+ *   affecting mask mode and inc, possibly some external modes)
+ */
```

Comment:
I think most of the comments above are remnants from `subsets.c` which you used as a template. Anything not applicable here should be dropped.